squidb/src/com/yahoo/squidb/data/ViewModel.java

```diff
@@ -184,20 +210,46 @@ protected static void validateAliasedProperties(Property<?>[] aliasedPropertyArr
 
     protected static class TableMappingVisitors {
 
-        private Map<Class<? extends AbstractModel>, TableModelMappingVisitor<?>> map =
-                new HashMap<Class<? extends AbstractModel>, TableModelMappingVisitor<?>>();
+        private Map<Class<? extends AbstractModel>, Map<SqlTable<?>, TableModelMappingVisitor<?>>> map =
+                new HashMap<Class<? extends AbstractModel>, Map<SqlTable<?>, TableModelMappingVisitor<?>>>();
 
-        private <T extends AbstractModel> void put(Class<T> cls, TableModelMappingVisitor<T> mapper) {
-            map.put(cls, mapper);
+        private <T extends AbstractModel> void put(Class<T> cls, SqlTable<?> table,
+                TableModelMappingVisitor<T> mapper) {
+            Map<SqlTable<?>, TableModelMappingVisitor<?>> visitors = map.get(cls);
+            if (visitors == null) {
+                visitors = new HashMap<SqlTable<?>, TableModelMappingVisitor<?>>();
+                map.put(cls, visitors);
+            }
+            visitors.put(table, mapper);
         }
 
         @SuppressWarnings("unchecked")
-        public <T extends AbstractModel> TableModelMappingVisitor<T> get(Class<T> cls) {
-            return (TableModelMappingVisitor<T>) map.get(cls);
+        public <T extends AbstractModel> TableModelMappingVisitor<T> get(Class<T> cls, SqlTable<?> table) {
+            Map<SqlTable<?>, TableModelMappingVisitor<?>> visitors = map.get(cls);
+            if (visitors == null) {
+                return null;
+            }
+            if (table == null) {
+                if (visitors.size() == 1) {
+                    for (TableModelMappingVisitor<?> visitor : visitors.values()) {
+                        return (TableModelMappingVisitor<T>) visitor;
+                    }
+                } else {
+                    // Log an error and return null
+                    Log.e(SquidUtilities.LOG_TAG, "Attempted to mapToModel for class " + cls + ", but multiple" +
```

Comment:
Open question -- should this throw an exception instead of returning null? Might be better for tracking down usage errors.